### PR TITLE
add MuSig related bindings

### PR DESCRIPTION
**Pull request by michael1011**
Builds on top of #44 

This PR adds MuSig related bindings from secp256k1.

Update: works!

### CODE REVIEW

File: lib/main.c [hunk 1]
```diff
@@ -451,4 +453,201 @@ int ec_seckey_verify(const unsigned char *seckey)
...
+  }
+  free(ptrs);
+}
+
+#define RETURN_ON_ZERO              \
+  if (ret == 0)                     \
+  {                                 \
+    secp256k1_context_destroy(ctx); \
+    return ret;                     \
+  }
```
Comment (altafan) [inline lib/main.c:482]: not an expert of C, just to know, which `ctx` is destroyed by this macro?

File: test/fixtures/musig.json [hunk 2]
```diff
@@ -0,0 +1,107 @@
+{
```
Comment (altafan) [inline test/fixtures/musig.json:1]: did you take these test vectors from somewhere? If yes, can you add a reference for the source to the issue description?

Comment (michael1011) [reply test/fixtures/musig.json:1]: I just made them up, to have something to ensure that I didn't break existing functions while developing.
I'll for proper test cases

File: lib/main.c (hunk 1 above)
Comment (michael1011) [reply lib/main.c:482]: Depends on where you use it. Macros in C are just textual replacement. So whatever `ctx` is in scope when using the macro will be destroyed

**Comment by kilrau** [comment]
BUMP up for review :pray:

**Review by altafan — APPROVED** [review]
LGTM, thank you so much for the contribution! 

Can i just ask where did you take the fixtures for the unit tests? Did you generate them or you reused some of another lib?

**Pull request merged**